src/main/java/com/thealgorithms/sorts/RadixSort.java

```diff
@@ -2,61 +2,58 @@
 
 import java.util.Arrays;
 
-final class RadixSort {
+/**
+ * This class provides an implementation of the radix sort algorithm.
+ * It sorts an array of integers in increasing order.
+ */
+public final class RadixSort {
     private RadixSort() {
     }
 
-    private static int getMax(int[] arr, int n) {
-        int mx = arr[0];
-        for (int i = 1; i < n; i++) {
-            if (arr[i] > mx) {
-                mx = arr[i];
-            }
+    /**
+     * Sorts an array of integers using the radix sort algorithm.
+     *
+     * @param array the array to be sorted
+     * @return the sorted array
+     */
+    public static int[] sort(int[] array) {
```

Comment:
The current implementation crashes for negative inputs.

Reply:
Yes. But I will add this. But implementation will be more complicated

Reply:
When I wrote the initial comment, I thought it would be challenging to add support for negative numbers. However, the implementation for handling negative numbers has turned out to be quite straightforward. This is because the same radix sort logic used for positive numbers can be applied to negative numbers as well.

Thanks to your feedback on the pull request, the code is now highly modular, and each method is clearly defined and distinguishable.

Here is the part that handles negative numbers:

```
int[] negatives = Arrays.stream(array).filter(x -> x < 0).map(x -> -x).toArray();
if (negatives.length > 0) {
    radixSort(negatives);
    reverseAndNegate(negatives);
}

```
And here are the two methods involved:
```
private static void reverseAndNegate(int[] array) {
    for (int i = 0; i < array.length / 2; i++) {
        int temp = array[i];
        array[i] = -array[array.length - 1 - i];
        array[array.length - 1 - i] = -temp;
    }
    if (array.length % 2 != 0) {
        array[array.length / 2] = -array[array.length / 2];
    }
}

private static void mergeNegativeAndPositiveArrays(int[] negatives, int[] sortedArray, int[] positives) {
    System.arraycopy(negatives, 0, sortedArray, 0, negatives.length);
    System.arraycopy(positives, 0, sortedArray, negatives.length, positives.length);
}
```
These methods are well-separated and clear, ensuring the implementation is easy to understand.

However, if you feel it would be better to remove them, that's also acceptable.

Reply:
Basically this approach is based on the fact that all of the negative numbers are smaller than non-negative.

I would keep it simple and implement it only for non-negative numbers.

Reply:
@vil02 reverted